app/scripts/modules/core/src/presentation/formsy/components/ExpressionInput.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import * as spel2js from 'spel2js';
+import autoBindMethods from 'class-autobind-decorator';
+import { truncate } from 'lodash';
+
+import { FormsyComponent, IFormsyComponentProps, IFormComponentState } from '../FormsyComponent';
+import { Markdown } from '../../Markdown';
+
+import '../spel2js.templateParser';
+import { Modal } from 'react-bootstrap';
+
+export interface IExpressionInputProps extends IFormsyComponentProps {
+  placeholder?: string;
+  context?: object;
+  locals?: object;
+  markdown?: boolean;
+}
+
+export interface IExpressionInputState extends IFormComponentState {
+  showContextModal: boolean;
+  spelPreview: string;
+  spelError: {
```

Comment:
If you make this a type itself, you don't have to cast all the things to strings below.